edit/platform.sh

```diff
@@ -123,15 +123,69 @@ incremental_deploy () {
   fi
 }
 
+# Check if the given command available in shell, with mute output.
+# @param string command name, i.e git, curl etc.
+shell_command_exist() {
+  type ${1} > /dev/null 2>&1
+}
+
+# Get the source branch of given PR.
+# - This feature should apply to github https based repo ONLY
+# @param string pull request id
+# @param string git remote https url
+# @param string github PAT (Personal Access Token)
+get_pr_source_branch() {
+  set +e
+  # get function arguments
+  GIT=${1}
+  PR_ID=${2}
+  REMOTE_URL=${3}
+  PAT=${4}
+
+  GIT_HOST=$(echo ${REMOTE_URL} | awk -F/ '{print $3}')
+  if [ ${GIT_HOST} != "github.com" ]; then
+    echo 'upstream branch setup is only available for github'
+    return 2
+  fi
+  GIT_OWNER=$(echo ${REMOTE_URL} | awk -F/ '{print $4}')
+  GIT_REPO=$(echo ${REMOTE_URL} | awk -F/ '{print $5}' | cut -d'.' -f1)
+  if ! (shell_command_exist curl); then
+    echo '"curl" is required to set upstream branch'
+    return 3
+  fi
+  if ! (shell_command_exist jq); then
+    echo '"jq" is required to set upstream branch'
+    return 4
+  fi
+
+  # Get PR info using github API.
+  ${GIT} fetch origin
+  PR_INFO=$(curl \
+    -s -f -H "Accept: application/vnd.github+json" \
+    -H "Authorization: Bearer ${PAT}" \
+    https://api.github.com/repos/${GIT_OWNER}/${GIT_REPO}/pulls/${PR_ID})
+  RESULT=$?
+  if [ ${RESULT} -ne 0 ];then
+    echo 'Failed to get PR info, please check github Pull Request number and PAT(personal access token) are valid'
+    return 5
+  fi
+  # Get branch upstream from head ref
+  UPSTREAM=$(echo ${PR_INFO} | jq -r '.head.ref')
```

Comment:
What will happen if the pR is from a fork? We need to handle that case, even ifit's just to print a message that "push" is not allowed from fork PRs, and just fetch the PR ref in that case.

Reply:
cross repo PRs will not checkout with source branch, updated.